docs/docs/daft.md

```diff
@@ -0,0 +1,148 @@
+├╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
+│ 1992-01-02 ┆ 175366628  ┆ gular accoun                   │
+├╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
+│ 1992-01-02 ┆ 2186602151 ┆ blithely even                  │
+├╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
+│ 1992-01-02 ┆ 3937663654 ┆ ake boldly among the ideas. s… │
+├╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
+│ 1992-01-02 ┆ 2186781220 ┆ thely. slyly pending ideas ar… │
+├╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
+│ 1992-01-02 ┆ 3937999493 ┆  haggle at the regular, pen    │
+├╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
+│ 1992-01-02 ┆ 2186933061 ┆ ickly. slyly                   │
+├╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
+│ 1992-01-02 ┆ 3938167204 ┆ carefully silent instructions… │
+╰────────────┴────────────┴────────────────────────────────╯
+
+(Showing first 8 rows)
+```
+
+Any subsequent filter operations on the Daft `df` DataFrame object will be correctly optimized to take advantage of Iceberg features such as hidden partitioning and file-level statistics for efficient reads.
+
+``` py
+import datetime
+
+# Filter which takes advantage of partition pruning capabilities of Iceberg
+df = df.where(df["L_SHIPDATE"] > datetime.date(1993, 1, 1))
+df.show()
+```
+
+```
+╭────────────┬────────────┬────────────────────────────────╮
+│ L_SHIPDATE ┆ L_ORDERKEY ┆ L_COMMENT                      │                                                                        
+│ ---        ┆ ---        ┆ ---                            │
+│ Date       ┆ Int64      ┆ Utf8                           │
+╞════════════╪════════════╪════════════════════════════════╡
+│ 1993-01-02 ┆ 5695313125 ┆  slyly special p               │
+├╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
+│ 1993-01-02 ┆ 2701326853 ┆ ironic instru                  │
+├╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
+│ 1993-01-02 ┆ 5695313766 ┆ ly according                   │
+├╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
+│ 1993-01-02 ┆ 2701330720 ┆ y alongside of the blithely    │
+├╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
+│ 1993-01-02 ┆ 5695315200 ┆ ckly final foxes haggle car    │
+├╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
+│ 1993-01-02 ┆ 2701331524 ┆ ns doze slyly pending instruc… │
+├╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
+│ 1993-01-02 ┆ 5695317377 ┆ re about the ironic, silen     │
+├╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌┼╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌╌┤
+│ 1993-01-02 ┆ 2701342819 ┆ fully even pinto beans wa      │
+╰────────────┴────────────┴────────────────────────────────╯
+
+(Showing first 8 rows)
+```
+
+### Type compatibility
+
+Daft and Iceberg have compatible type systems. Here are how types are converted across the two systems.
+
+When reading from an Iceberg source into Daft:
```

Comment:
It might actually be different in the other direction, since the Daft types are a superset of the Iceberg types (e.g. we'll have to figure out how the Daft tensor types will interact with Iceberg). I can remove this for now until we have write support though.